applications/tari_base_node/src/parser.rs

```diff
@@ -175,4 +178,82 @@ impl Parser {
             },
         }
     }
+
+    // Function to process  the get balance command
+    fn process_get_balance(&mut self) {
+        let mut handler = self.base_node_context.wallet_output_service.clone();
+        self.executor.spawn(async move {
+            let result = handler.get_balance().await;
+            if result.is_err() {
+                println!("Something went wrong");
+                warn!(
+                    target: LOG_TARGET,
+                    "Error communicating with wallet: {}",
+                    result.err().unwrap().to_string(),
+                );
+                return;
+            };
+            println!("Current balance is: {:?}", result.unwrap());
+        });
+    }
+
+    // Function to process  the get chain meta data
+    fn process_get_chain_meta(&mut self) {
+        let mut handler = self.base_node_context.node_service.clone();
+        self.executor.spawn(async move {
+            let result = handler.get_metadata().await;
+            if result.is_err() {
+                println!("Something went wrong");
+                warn!(
+                    target: LOG_TARGET,
+                    "Error communicating with base node: {}",
+                    result.err().unwrap().to_string(),
+                );
+                return;
+            };
+            println!("Current meta data is is: {:?}", result.unwrap());
```

Comment:
We may as well implement `Display` for anything that doesn't already so the user gets nicer looking output

Reply:
done